### ISSUE TITLE
When schema type inferred, boolean enum values incorrectly deserialized as strings

### ISSUE THREAD

**Issue by kerrykimbrough** [swagger-api/swagger-parser#1129]
Consider the following schema:

```
{
  "enum": [ true, false]
}
```

When deserialized to a `Schema` object, the type is correctly inferred to be "boolean":

```
assertEquals(  schema.getType(), "boolean"); // Passes
```

But "enum" values are not deserialized correctly:

```
assertEquals( schema.getEnum().get(0), "true"); // Passes!
assertEquals( schema.getEnum().get(1), Boolean.FALSE); // Fails!
```

**Comment by kerrykimbrough** [comment]
This defect is demonstrated and repaired by changes in PR #1130

**Comment by gracekarina** [comment]
this issue was closed by #1130
